File: src/modules/presentation/io/tripled/social/client/presentation/controller/WallController.java
package io.tripled.social.client.presentation.controller;

import io.tripled.social.client.application.ReadWallUseCase;
import io.tripled.social.client.presentation.Controller;
import io.tripled.social.client.presentation.Output;
import io.tripled.social.client.presentation.request.WallCliRequest;

public class WallController implements Controller<WallCliRequest> {
  private ReadWallUseCase readWallUseCase;

  public WallController(ReadWallUseCase readWallUseCase) {
    this.readWallUseCase = readWallUseCase;
  }

  @Override
  public void execute(WallCliRequest request, Output output) {
    output.print(readWallUseCase.readWall(request.getUserName()));
  }
}

File: src/main/kotlin/io/tripled/social/client/infrastructure/cli/controller/WallController.kt
package io.tripled.social.client.infrastructure.cli.controller

import io.tripled.social.client.application.ReadWall
import io.tripled.social.client.infrastructure.cli.Controller
import io.tripled.social.client.infrastructure.cli.Output
import io.tripled.social.client.infrastructure.cli.request.WallCliRequest

class WallController(private val readWall: ReadWall) : Controller<WallCliRequest> {
    override fun execute(request: WallCliRequest, output: Output) {
        output.print(readWall.readWall(request.userName))
    }
}
